presto-password-authenticators/src/main/java/io/prestosql/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -290,6 +291,14 @@ private DirContext createUserDirContext(String userDistinguishedName, String pas
                 .build();
     }
 
+    private static void validateLdapUrl(String ldapUrl, boolean allowInsecure)
+    {
+        if (ldapUrl.matches("ldap://.*")) {
+            checkState(allowInsecure, "Connecting to the LDAP server without SSL enabled requires `ldap.allow-insecure=true`");
```

Comment:
Add this as an `@AssertTrue` validation the config class (search code base for examples)

This should be `checkArgument` since it's validating an argument. `IllegalStateException` means the class is in an inappropriate state for the current method invocation.